nilearn/_utils/cache_mixin.py

```diff
@@ -80,17 +80,18 @@ def _check_memory(memory, verbose=0):
     return memory
 
 
+def _nibabel_versions_equal(versions, my_versions):
```

Comment:
as this is not necessarily specific to nibabel and applies to a pair of version, not collections, maybe we can change the names a little bit eg `_safe_versions_equal(version_a, version_b)`

Reply:
I would suggest a few tests for this function with the name suggested by @jeromedockes :
```
diff --git a/nilearn/tests/test_cache_mixin.py b/nilearn/tests/test_cache_mixin.py
index 7687b37e..615223e7 100644
--- a/nilearn/tests/test_cache_mixin.py
+++ b/nilearn/tests/test_cache_mixin.py
@@ -182,3 +182,19 @@ def test_cache_shelving():
         res = cache_mixin.cache(f, mem, shelve=True)(2)
         assert res.get() == 2
         assert len(_get_subdirs(joblib_dir)) == 1
+
+
+@pytest.mark.parametrize('version_a,version_b,expected', [
+    ("1.2", "1.2", True),
+    ("1.2.a.dev", "1.2a0dev", True),
+    ("1.2", "1.3", False),
+    ("1.2.3", "1.2", False),
+    ("1.2", "1.2.3", False),
+    (None, "1.2", False),
+    (None, None, False),
+    ("1.2", None, False),
+    ("1.2", 1, False),
+    ("1.2", {}, False),
+])
+def test_safe_versions_equal(version_a, version_b, expected):
+    assert cache_mixin._safe_versions_equal(version_a, version_b) == expected
```
I also think this is the correct unit where to test the `TypeError` as discussed [here](https://github.com/nilearn/nilearn/pull/3375/files#r986539527).